spec/acceptance/should_create_key_spec.rb

```diff
@@ -0,0 +1,145 @@
+require 'tempfile'
+require 'pathname'
+require 'spec_helper_acceptance'
+
+describe 'Registry Key Management' do
+
+  # Setup the master to use the modules specified in the --modules option
+
+  context 'registry test key variable setup' do
+
+    # Generate a unique key name
+    keyname = "PuppetLabsTest_#{random_string(8)}"
+    # This is the keypath we'll use for this entire test.  We will actually create this key and delete it.
+    keypath = "HKLM\\Software\\Vendor\\#{keyname}"
+
+    # Setup keys to purge
+    phase1 = <<PHASE1
+  registry_key { 'HKLM\\Software\\Vendor': ensure => present }
+  Registry_key { ensure => present }
+  Registry_value { ensure => present, data => 'Puppet Default Data' }
+
+  registry_key { '#{keypath}': }
+  registry_key { '#{keypath}\\SubKey1': }
+
+  if $architecture == 'x64' {
+    registry_key { '32:#{keypath}': }
+    registry_key { '32:#{keypath}\\SubKey1': }
+  }
+
+  registry_key   { '#{keypath}\\SubKeyToPurge': }
+  registry_value { '#{keypath}\\SubKeyToPurge\\Value1': }
+  registry_value { '#{keypath}\\SubKeyToPurge\\Value2': }
+  registry_value { '#{keypath}\\SubKeyToPurge\\Value3': }
+
+  if $architecture == 'x64' {
+    registry_key   { '32:#{keypath}\\SubKeyToPurge': }
+    registry_value { '32:#{keypath}\\SubKeyToPurge\\Value1': }
+    registry_value { '32:#{keypath}\\SubKeyToPurge\\Value2': }
+    registry_value { '32:#{keypath}\\SubKeyToPurge\\Value3': }
+  }
+PHASE1
+
+    # Purge the keys in a subsequent run
+    phase2 = <<PHASE2
+  registry_key { 'HKLM\\Software\\Vendor': ensure => present }
+  Registry_key { ensure => present, purge_values => true }
+
+  registry_key { '#{keypath}\\SubKeyToPurge': }
+  if $architecture == 'x64' {
+    registry_key { '32:#{keypath}\\SubKeyToPurge': }
+  }
+PHASE2
+
+    # Delete our keys
+    phase3 = <<PHASE3
+  registry_key { 'HKLM\\Software\\Vendor': ensure => present }
```

Comment:
Not sure if this is super important, but I'm seeing odd whitespace and indenting here.